### PR TITLE
Re-enable flake8 for core package and tests

### PR DESCRIPTION
**Pull request by jonparrott**
Note to reviewers: this is intentionally 2 commits. The first re-enables linting, the second addresses lint issues.

### CODE REVIEW

File: core/google/cloud/_helpers.py [hunk 1]
```diff
@@ -25,21 +25,21 @@
...
+from six.moves import http_client
+
 import google.auth
 from google.protobuf import duration_pb2
 from google.protobuf import timestamp_pb2
-import google_auth_httplib2
 
 try:
     import grpc
     import google.auth.transport.grpc
```
Comment (dhermes) [inline core/google/cloud/_helpers.py]: Does `flake8` not care about these?

File: core/google/cloud/_http.py [hunk 2]
```diff
@@ -16,8 +16,8 @@
 
 import json
 import platform
-from pkg_resources import get_distribution
 
+from pkg_resources import get_distribution
```
Comment (dhermes) [inline core/google/cloud/_http.py]: TIL! Which 3rd party package does this come from? `setuptools`?

Comment (jonparrott) [inline core/google/cloud/_http.py]: yep, setuptools.

File: core/google/cloud/_helpers.py (hunk 1 above)
Comment (jonparrott) [inline core/google/cloud/_helpers.py]: It's smart enough to know that try: except imports are their own beasties.

**Pull request merged**